### PR TITLE
vulkaninfo: Correctly query per-surface queue support

### PR DESCRIPTION
**Pull request by charles-lunarg**
Plus other small improvements.

### CODE REVIEW

File: vulkaninfo/vulkaninfo.cpp [hunk 1]
```diff
@@ -396,17 +396,19 @@ void GpuDumpQueueProps(Printer &p, AppGpu &gpu, const AppQueueFamilyProperties &
...
-        p.PrintKeyString("present support", queue.platforms_support_present ? "true" : "false");
+    if (!queue.can_present) {
+        p.PrintKeyString("present support", "false");
+    } else if (queue.can_always_present) {
+        p.PrintKeyString("present support", "true");
     } else {
         size_t width = 0;
-        for (auto &surface : gpu.inst.surface_extensions) {
-            if (surface.name.size() > width) width = surface.name.size();
+        for (auto &support : queue.present_support) {
```
Comment (ziga-lunarg) [inline vulkaninfo/vulkaninfo.cpp:405]: Could support be const for clarity?

File: vulkaninfo/vulkaninfo.h [hunk 2]
```diff
@@ -243,7 +243,7 @@ auto GetVectorInit(const char *func_name, F &&f, T init, Ts &&...ts) -> std::vec
         err = f(ts..., &count, results.data());
         results.resize(count);
         iteration_count++;
-    } while (err == VK_INCOMPLETE || iteration_count < max_iterations);
+    } while (err == VK_INCOMPLETE || iteration_count >= max_iterations);
```
Comment (ziga-lunarg) [inline vulkaninfo/vulkaninfo.h:246]: What is the reason for this change? It looks wrong, could lead to infinite loop

Comment (charles-lunarg) [reply vulkaninfo/vulkaninfo.h:246]: The reason for the change was that the loop was always run 3 times. I want the loop to only run once if necessary, but I can see how it would lead to infinite loops now.

I'll have to rethink the logic some more.

**Review by ziga-lunarg — APPROVED** [review]
